scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +5217,149 @@ def __rtruediv__(self, other):
 
     def __neg__(self):
         return self * -1
+
+
+class IMFTransformedDistribution(TransformedDistribution):
+    r"""Distribution underlying a strictly monotonic function of a random variable
+
+    Given a random variable :math:`X`; a strictly monotonic function
+    :math:`g(u)`, its inverse :math:`h(u) = g^{-1}(u)`, and the derivative magnitude
+    :math: `|h'(u)| = \left| \frac{dh(u)}{du} \right|`, define the distribution
+    underlying the random variable :math:`Y = g(X)`.
+
+    Parameters
+    ----------
+    X : `ContinuousDistribution`
+        The random variable :math:`X`.
+    g, h, dh : callable
+        Elementwise functions representing the mathematical functions
+        :math:`g(u)`, :math:`h(u)`, and :math:`|h'(u)|`
+    logdh : callable, optional
+        Elementwise function representing :math:`\log(h'(u))`.
+        The default is ``lambda u: np.log(dh(u))``, but providing
+        a custom implementation may avoid over/underflow.
+    increasing : bool, optional
+        Whether the function is strictly increasing (True, default)
+        or strictly decreasing (False).
+
+    """
+
+    def __init__(self, dist, *args, g, h, dh, logdh=None, increasing=True, **kwargs):
+        super().__init__(dist, *args, **kwargs)
```

Comment:
We need to overload  `__repr__`.  Right now it just uses the __repr__ for `dist`

```python
import scipy.stats as stats

X = stats.Normal(mu=0, sigma=1)
Y = stats.exp(Y)
print(Y)
# Normal(mu=0.0, sigma=1.0)
```

It's not obvious what to do though. Since what is now called `IMFTransformedDistribution` is intended to be private, we can get away with passing a function or data needed for  `__repr__` as an argument to `__init__`. 

Something like this is likely the best we can do, since the `reprs` for `g`, `h`, etc. will be opaque things like `<function <lambda> at 0x714ea2b96700>`.  Even then, we'd be confounded by inconsistent naming conventions for transformed distributions, like you pointed out for `lognormal` vs `loggamma`, so even for specific transforms, there might not be a clean thing to do.

For now, let's just think about what to do for the specific cases in this PR.

[Inverse Distribution](https://en.wikipedia.org/wiki/Inverse_distribution) is pretty standard for the reciprocal of a random variable, so in that case, i think it makes sense to make the `repr` `Inv<something>`.

For `exp`, `"exp(Normal(mu=0.0, sigma=1.0)"` is confusing in my opinion. 

Something like

`"exp(X): X ~ Normal(mu=0.0, sigma=1.0)"`

is more clear, but a bit verbose, and it's not clear that X is being used more like a free variable, and doesn't refer to any particular X.

`"exp(*): * ~ Normal(mu=0.0, sigma=1.0)"`  perhaps clears up the issue that we don't mean a particular X or whatever, but is confusing in its own right.

Any thoughts?

Reply:
Good catch. To put it in perspective, there is a very low bar to meet. For a few decades, frozen instances of `rv_continuous` have looked like:

```
import scipy
scipy.stats.norm(1, 2)
# <scipy.stats._distn_infrastructure.rv_continuous_frozen object at 0x796e4c735090>
```

regardless of the underlying distribution family, and apparently nobody has complained. Of course, I'd like to do better; I just thought it was worth mentioning where we're coming from.

Re: 

> For exp, "exp(Normal(mu=0.0, sigma=1.0))" is confusing in my opinion.

Why? That is actually a valid `repr` (i.e. that is actually code to reproduce the object), and it's pretty readable. Since the standards for naming the distributions underlying transformations of random variables are not very rigid, I don't think we're going to be able to do something like `LogNormal(mu=0.0, sigma=1.0)` in general, so I think showing how the random variable itself is transformed is the best thing to do.

`inv(Normal(mu=0.0, sigma=1.0))` works, but I didn't plan on exposing an `inv` function, and we're probably going to add powers, after this and the `abs` PR are done, so maybe it should be `(Normal(mu=0.0, sigma=1.0))**(-1)`?

Reply:
I guess `exp(Normal(mu=0.0, sigma=1.0))` is fine. I was thinking in terms of distributions, not in terms of Python objects, it makes it look like we're taking `exp` of a distribution. `1 / Normal(mu=0.0, sigma=1.0)` seems fine for division. I guess the `repr` needs to not just be a string representation, but a way to generate an instance from a string right? In which case those seem like the best options.

Having an opaque thing analogous to `# <scipy.stats._distn_infrastructure.rv_continuous_frozen object at 0x796e4c735090>
` is fine too, if not ideal, and could be improved later. Having it be the `repr` for a different distribution is wading into bug territory though imo, and should be fixed in some way.

Reply:
It's definitely better, but it would be good if it had enough info to recover what the transform was if we're making these public. It shouldn't be too much work to get things like `exp(Normal(mu=0.0, sigma=1.0))` and `1/Gamma(a=2)`.

Reply:
Done.